wallet/src/de/schildbach/wallet/ui/AdvancedSecurityActivity.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2019 Dash Core Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.schildbach.wallet.ui
+
+import android.os.Bundle
+import android.text.SpannableStringBuilder
+import android.text.style.ImageSpan
+import android.view.View
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import androidx.core.content.res.ResourcesCompat
+import de.schildbach.wallet_test.R
+import kotlinx.android.synthetic.main.activity_advanced_security.*
+import java.util.concurrent.TimeUnit
+
+
+enum class SecurityLevel {
+    NONE, LOW, MEDIUM, HIGH, VERY_HIGH
+}
+
+class AdvancedSecurityActivity : BaseMenuActivity() {
+    private lateinit var dashSymbol: ImageSpan
+
+    override fun getLayoutId(): Int {
+        return R.layout.activity_advanced_security
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val drawableDash = ResourcesCompat.getDrawable(resources,
+                R.drawable.ic_dash_d_black, null)
+        drawableDash!!.setBounds(0, 0, 32, 32)
+        dashSymbol = ImageSpan(drawableDash, ImageSpan.ALIGN_BASELINE)
+
+        auto_logout_switch.setOnCheckedChangeListener { _, enabled ->
+            configuration.autoLogoutEnabled = enabled
+            updateView()
+        }
+
+        spending_confirmation_switch.setOnCheckedChangeListener { _, enabled ->
+            configuration.spendingConfirmationEnabled = enabled
+            updateView()
+        }
+
+        auto_logout_seekbar.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
```

Comment:
Can we move these listeners to class fields? This way the `onCreate` method will be shorter and easier to read.